### PR TITLE
Merge index management to index-operation-common

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
Signed-off-by: suzhou <suzhou@amazon.com>

### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by xluo-aws — APPROVED** [review]

File: public/pages/CreateIndex/components/IndexDetail/IndexDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,548 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Ref, forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState } from "react";
```
Comment (amsiglan) [inline public/pages/CreateIndex/components/IndexDetail/IndexDetail.tsx:6]: Not blocking for this PR, but please break down this component into smaller files. Also try moving interfaces, constants to separate files

File: public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+                            target="_blank"
+                            external
+                            href={`https://opensearch.org/docs/${coreService?.docLinks.DOC_LINK_VERSION}/opensearch/mappings/`}
+                          >
+                            Learn more.
+                          </EuiLink>
+                        </div>
+                      }
+                    >
+                      <></>
```
Comment (amsiglan) [inline public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx:348]: empty fragment.

**Review by amsiglan — APPROVED** [review]
Changes look good overall. Please split big components into multiple files and move interfaces/helpers to their own files

File: public/pages/CreateIndex/components/AliasSelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    <RemoteSelect
+      {...(props as Partial<EuiComboBoxProps<any>>)}
+      onOptionsChange={(options) => {
+        optionsRef.current = options;
+        onOptionsChange?.(options);
+      }}
+      placeholder="Select aliases or input new aliases."
+      customOptionText="Add {searchValue} as a new alias."
+      refreshOptions={refreshOptions}
+      value={transformObjToArray(value).map((item) => item.label)}
```
Comment (awick) [inline public/pages/CreateIndex/components/AliasSelect/index.tsx:59]: Is there a reason this isn't just Object.keys(value)? It looks like this removes the label that transformObjToArray adds

File: public/pages/CreateIndex/components/IndexMapping/IndexMapping.tsx [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+      fieldName,
+    };
+    if (properties) {
+      payload.properties = transformObjectToArray(properties);
+    }
+    return payload;
+  });
+};
+
+export const transformArrayToObject = (array: MappingsProperties): MappingsPropertiesObject => {
```
Comment (awick) [inline public/pages/CreateIndex/components/IndexMapping/IndexMapping.tsx:30]: Kind of confusing there is both a transformArrayToObject and transformArrayToObj. Maybe add some comments?

**Review by awick — COMMENTED** [review]
I'm not super familiar with the technology here, so not many comments. I would like to see
* comments on any utility/shared functions
* when doing decomposition keep functions closer together

File: public/pages/CreateIndex/components/IndexDetail/IndexDetail.tsx (hunk 1 above)
Comment (SuZhou-Joe) [reply public/pages/CreateIndex/components/IndexDetail/IndexDetail.tsx:6]: You are right, the file has over 500 lines, will refractor that as a TODO.

File: public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply public/pages/IndexDetail/containers/IndexDetail/IndexDetail.tsx:348]: Same for typescript type check.

File: public/pages/CreateIndex/components/AliasSelect/index.tsx (hunk 3 above)
Comment (SuZhou-Joe) [reply public/pages/CreateIndex/components/AliasSelect/index.tsx:59]: Good point, fixed on https://github.com/opensearch-project/index-management-dashboards-plugin/commit/3852583062f4d60385539ddd977ea1f310ab9291

File: public/pages/CreateIndex/components/IndexMapping/IndexMapping.tsx (hunk 4 above)
Comment (SuZhou-Joe) [reply public/pages/CreateIndex/components/IndexMapping/IndexMapping.tsx:30]: Good point, fixed on https://github.com/opensearch-project/index-management-dashboards-plugin/commit/3852583062f4d60385539ddd977ea1f310ab9291

**Pull request closed without merging**